common/src/objects/workload_state.rs

```diff
@@ -18,82 +18,360 @@ use serde::{Deserialize, Serialize};
 
 use api::proto;
 
-// [impl->swdd~common-supported-workload-states~1]
-#[derive(Debug, Default, PartialEq, Eq, Clone, Serialize, Deserialize)]
-pub enum ExecutionState {
-    #[default]
-    ExecUnknown = 0,
-    ExecPending = 1,
-    ExecWaitingToStart = 2,
-    ExecStarting = 3,
-    ExecRunning = 4,
-    ExecSucceeded = 5,
-    ExecFailed = 6,
-    ExecWaitingToStop = 7,
-    ExecStopping = 8,
-    ExecRemoved = 10,
-}
-
-impl From<i32> for ExecutionState {
+use crate::std_extensions::UnreachableOption;
+
+use super::WorkloadExecutionInstanceName;
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub enum PendingSubstate {
+    Initial = 0,
```

Comment:
I cannot see that conversion to `Initial` is unit tested and that the state is set to initial in production code at the beginning.